File: backend/src/main/java/VoidSystems/appointment_service/service/ServiceService.java
package VoidSystems.appointment_service.service;

import java.util.List;

import VoidSystems.appointment_service.dto.service.ServiceDTO;

public interface ServiceService {
    
    List<ServiceDTO> getAllServices();
    
    List<ServiceDTO> getAllActiveServices();
    
    List<ServiceDTO> getServicesByProviderId(Long providerId);
    
    List<ServiceDTO> getActiveServicesByProviderId(Long providerId);
    
    ServiceDTO getServiceById(Long id);
    
    ServiceDTO getServiceByIdAndProviderId(Long id, Long providerId);
    
    ServiceDTO createService(ServiceDTO serviceDTO, Long providerId);
    
    ServiceDTO updateService(Long id, ServiceDTO serviceDTO);
    
    void deleteService(Long id);
    
    void activateService(Long id);
    
    void deactivateService(Long id);
}

File: frontend/src/api/serviceService.ts
import { ApiResponse, Service } from '../types';
import axiosInstance from './axiosConfig';

export const serviceService = {
  getAllServices: async (activeOnly?: boolean): Promise<Service[]> => {
    const response = await axiosInstance.get<ApiResponse<Service[]>>('/services', {
      params: { activeOnly },
    });
    return response.data.data;
  },

  getServiceById: async (id: number): Promise<Service> => {
    const response = await axiosInstance.get<ApiResponse<Service>>(`/services/${id}`);
    return response.data.data;
  },

  getServicesByProviderId: async (providerId: number, activeOnly?: boolean): Promise<Service[]> => {
    const response = await axiosInstance.get<ApiResponse<Service[]>>(`/services/provider/${providerId}`, {
      params: { activeOnly },
    });
    return response.data.data;
  },

  createService: async (providerId: number, serviceData: Partial<Service>): Promise<Service> => {
    const response = await axiosInstance.post<ApiResponse<Service>>(`/services/provider/${providerId}`, serviceData);
    return response.data.data;
  },

  updateService: async (id: number, serviceData: Partial<Service>): Promise<Service> => {
    const response = await axiosInstance.put<ApiResponse<Service>>(`/services/${id}`, serviceData);
    return response.data.data;
  },

  deleteService: async (id: number): Promise<void> => {
    await axiosInstance.delete<ApiResponse<void>>(`/services/${id}`);
  },

  activateService: async (id: number): Promise<void> => {
    await axiosInstance.put<ApiResponse<void>>(`/services/${id}/activate`);
  },

  deactivateService: async (id: number): Promise<void> => {
    await axiosInstance.put<ApiResponse<void>>(`/services/${id}/deactivate`);
  },
};